File: SoftUni-Algorithms-April-2016/Exam Preparation/Sample Exam/1. Sequences-of-Limited-Sum/SequencesOfLimitedSum.java
import java.util.Scanner;

public class SequencesOfLimitedSum {
	static StringBuilder allSequences = new StringBuilder();
	static int[] seq;

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int maxSum = scan.nextInt();
		seq = new int[maxSum];
        generateSequences(maxSum, 0);
		System.out.println(allSequences);
	}

	static void generateSequences(int maxSum, int index) {
		for (int num = 1; num <= maxSum; num++) {
			seq[index] = num;
			if (maxSum >= 0) {
				// Print seq[0...index]
				for (int i = 0; i <= index; i++) {
					allSequences.append(seq[i]);
					if (i < index) {
						allSequences.append(" ");
					}
				}
				allSequences.append("\n");
			}
			generateSequences(maxSum - num, index + 1);
		}
	}
}

File: SoftUni-Algorithms-April-2016/Exam Preparation/Sample Exam/1. Sequences-of-Limited-Sum/SequencesOfLimitedSum.cs
using System;
using System.Text;

public class SequencesOfLimitedSum
{
	static StringBuilder allSequences = new StringBuilder();
	static int[] seq;

	static void Main()
	{
		int maxSum = int.Parse(Console.ReadLine());
        seq = new int[maxSum];
        GenerateSequences(maxSum, 0);
		Console.Write(allSequences);
	}

	static void GenerateSequences(int maxSum, int index)
	{
		for (int num = 1; num <= maxSum; num++)
		{
            seq[index] = num;
			if (maxSum >= 0)
			{
                // Print seq[0...index]
                for (int i = 0; i < index; i++)
                {
                    allSequences.Append(seq[i]);
                    allSequences.Append(" ");
                }
                allSequences.Append(seq[index]);
                allSequences.AppendLine();
            }
			GenerateSequences(maxSum - num, index + 1);
		}
	}
}
